### ISSUE TITLE
Improve version number sorting (use publish date if possible)

### ISSUE THREAD

**Issue by scovetta** [microsoft/OSSGadget#31]
Example: pkg:composer/ircmaxell/random-lib

Result: Gets `dev-improved_string_presets`.
Should: Get v1.2.0

**Comment by jacobmsft** [comment]
Do you mean sorting the versions of a package? I have included the SemanticVersioning nuget if that's the case.

**Comment by scovetta** [comment]
Yep, the problem is that not all package management systems are semver-compliant -- take this one for example (https://packagist.org/packages/ircmaxell/random-lib).
